### PR TITLE
NMS-13837 Add Health Check Rest API on Sentinel

### PR DESCRIPTION
**Pull request by aramosvizcarra**
- Reused the Minion Rest Health check
- Moved Minion Rest from features/minion/core/rest into core/health/rest-cxf
- Both Minion and Sentinel now use the same Health Rest bundle feature

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-13837

### CODE REVIEW

File: container/features/src/main/resources/features.xml
Comment (cgorantla) [inline container/features/src/main/resources/features.xml:228]: Is this needed ? Looks like wrap is already part of standard karaf

File: features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg
Comment (cgorantla) [inline features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg:57]: If we need these features for `opennms-health-rest-service` to work, we need them to be part that feature instead of adding them here.

File: features/sentinel/repository/pom.xml
Comment (cgorantla) [inline features/sentinel/repository/pom.xml:62]: Do we need `cxf-commands` ?    We also need to add `opennms-health-rest-service`  and new dependency in minion/repository/pom.xml as well.

**Review by cgorantla — CHANGES_REQUESTED** [review]

File: features/sentinel/repository/pom.xml
Comment (aramosvizcarra) [reply features/sentinel/repository/pom.xml:62]: - Included Karaf features in () under org.apache.karaf.features.cfg so there are no missing dependencies at boot.
- Rearrange health rest features to include required features.
- cfx-commands was already included in minion features (minion/resources/pom.xml)

File: container/features/src/main/resources/features.xml
Comment (cgorantla) [inline container/features/src/main/resources/features.xml:1847]: Need to check if we really need `org.json` and `owasp-encoder` features here.

File: features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg
Comment (cgorantla) [inline features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg:36]: Think we shouldn't need to change any of this grouping.

File: features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg
Comment (cgorantla) [inline features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg:58]: Since we are adding these two features by default on Sentinel, we should try to run full smoke suite before merging.  You need to push another branch with `smoke` in the branch name like `AR/jira/NMS-13837-smoke`. That will trigger full smoke suite run.  Once that passes we can merge this.

File: features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg
Comment (aramosvizcarra) [reply features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg:36]: I think this is only necessary because the wrap feature needs to be installed before the custom features, that's why I added earlier wrap under owsap-encoder feature.
Otherwise I start getting an exception when running runInPlace.sh 
java.lang.IllegalStateException: Unknown protocol: wrap
at
org.apache.felix.framework.URLHandlersStreamHandlerProxy.parseURL

File: features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg
Comment (aramosvizcarra) [reply features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg:58]: Agree

File: features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg
Comment (cgorantla) [reply features/container/sentinel/src/main/filtered-resources/etc/org.apache.karaf.features.cfg:36]: okay, let me reproduce locally.

File: container/features/src/main/resources/features.xml
Comment (cgorantla) [reply container/features/src/main/resources/features.xml:1847]: Looks like you only need`org.json` here, don't need `owasp-encoder`

**Comment by cgorantla** [comment]
Looks good to me. Let's wait for full smoke to pass.

**Review by cgorantla — APPROVED** [review]
LGTM, please merge once full smoke test passes

**Review by cgorantla — APPROVED** [review]

**Review by cgorantla — APPROVED** [review]

**Comment by cgorantla** [comment]
Smoke tests are passing https://app.circleci.com/pipelines/github/OpenNMS/opennms/15632/workflows/442d980a-6110-4d02-9620-dcfe443486d1.

Merging.

**Pull request merged**
